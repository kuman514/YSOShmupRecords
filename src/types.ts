export interface NavNodeInfo {
  id: string;
  label: string;
  childNavNodeIds: NavNodeInfo['id'][];
}

export interface ShmupRecord {
  id: string;
  when: Date;
  subjectId: string;
  stage: string;
  /**
   * Why score should be string?
   * In case of exceeding `Number.MAX_SAFE_INTEGER`, which equals to `9007199254740991`.
   */
  score: string;
  byWhat: string;
  comment: string;
  thumbnailUrl: string;
  originalImageUrl: string;
  tweetUrl: string;
  youtubeUrl?: string;
  specialTags?: string[];
}
