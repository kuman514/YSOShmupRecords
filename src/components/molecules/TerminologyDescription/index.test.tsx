import { cleanup, render } from '@testing-library/react';
import 'jest-styled-components';
import { beforeEach, describe, expect, it } from 'vitest';

import { Terminology } from '^/types';

import { TerminologyDescription } from '.';

const rootDescriptionListItems: Terminology[] = [
  {
    id: 'test-1',
    term: 'term 1',
    description: 'Root Test 1',
  },
  {
    id: 'test-1-1',
    term: 'term 1-1',
    description: 'Root Test Sub 1',
  },
  {
    id: 'test-1-2',
    term: 'term 1-2',
    description: 'Root Test Sub 2',
  },
  {
    id: 'test-1-2-1',
    term: 'term 1-2-1',
    description: 'Root Test Sub Sub 1',
  },
  {
    id: 'test-1-2-2',
    term: 'term 1-2-2',
    description: 'Root Test Sub Sub 2',
  },
  {
    id: 'test-1-2-3',
    term: 'term 1-2-3',
    description: 'Root Test Sub Sub 3',
  },
  {
    id: 'test-1-3',
    term: 'term 1-3',
    description: 'Root Test Sub 3',
  },
  {
    id: 'test-2',
    term: 'term 2',
    description: 'Root Test 2',
  },
  {
    id: 'test-3',
    term: 'term 3',
    description: 'Root Test 3',
  },
  {
    id: 'test-3-1',
    term: 'term 3-1',
    description: 'Root Test Sub 5',
  },
];

describe('TerminologyDescription', () => {
  beforeEach(() => {
    cleanup();
  });

  it('should have all root list items and their sub-items (match snapshot)', async () => {
    const { container } = render(
      <TerminologyDescription
        title="Test 1 kuman514"
        descriptionListItems={rootDescriptionListItems}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
