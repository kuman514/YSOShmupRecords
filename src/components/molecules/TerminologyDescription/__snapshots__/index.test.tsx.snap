// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TerminologyDescription > should have all root list items and their sub-items (match snapshot) 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  row-gap: 16px;
  padding-left: 15px;
  padding-right: 15px;
}

.c1 {
  font-size: 36px;
  font-weight: 700;
}

.c2 {
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c3 {
  display: flex;
  flex-direction: column;
  gap: 4px;
}

.c4 {
  font-size: 18px;
  font-weight: 700;
}

.c5 {
  font-size: 16px;
  font-weight: 400;
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    Test 1 kuman514
  </h1>
  <ul
    class="c2"
  >
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 1
        </span>
        <span
          class="c5"
        >
          Root Test 1
        </span>
      </div>
    </li>
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 1-1
        </span>
        <span
          class="c5"
        >
          Root Test Sub 1
        </span>
      </div>
    </li>
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 1-2
        </span>
        <span
          class="c5"
        >
          Root Test Sub 2
        </span>
      </div>
    </li>
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 1-2-1
        </span>
        <span
          class="c5"
        >
          Root Test Sub Sub 1
        </span>
      </div>
    </li>
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 1-2-2
        </span>
        <span
          class="c5"
        >
          Root Test Sub Sub 2
        </span>
      </div>
    </li>
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 1-2-3
        </span>
        <span
          class="c5"
        >
          Root Test Sub Sub 3
        </span>
      </div>
    </li>
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 1-3
        </span>
        <span
          class="c5"
        >
          Root Test Sub 3
        </span>
      </div>
    </li>
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 2
        </span>
        <span
          class="c5"
        >
          Root Test 2
        </span>
      </div>
    </li>
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 3
        </span>
        <span
          class="c5"
        >
          Root Test 3
        </span>
      </div>
    </li>
    <li>
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          term 3-1
        </span>
        <span
          class="c5"
        >
          Root Test Sub 5
        </span>
      </div>
    </li>
  </ul>
</div>
`;
